### ISSUE TITLE
`FUSE` (or similar) support on `Windows`

### ISSUE THREAD

**Issue by radumarias** [xoriors/rencfs#3]
- https://crates.io/crates/winfsp
- https://github.com/SnowflakePowered/winfsp-rs

see #9 

A very basic working and tested implementation is needed

- We need to create smth similar to https://github.com/radumarias/rencfs/blob/66bbbf25528f51689c44bd3a51561acac7ad198c/src/mount/linux.rs
- Which wraps smth like `EncryptedFsFuse3` which will implement the interface between `WinFSP` and `EncryptedFs`
- Need to see specifics on how to use `WinFSP`

They don't really have examples and the docs are very scarce, but seems it's very used https://github.com/winfsp/winfsp/wiki/Known-File-Systems so it could be good for us too. Maybe you can find some examples in C or other langs if any of the dependants frm wiki are OSS.

There's these too https://github.com/billziss-gh/winfuse but I didn't found any bindings. `WinFSP` seems the most used, recommended and stable.

**Comment by opiredev** [comment]
@radumarias created a $42.00 reward using [Opire](https://opire.dev) 
 <details><summary>How to earn this reward?</summary>

Since this project does not have Opire bot installed yet 😞 you need to go to [Opire](https://app.opire.dev) and claim the rewards through your programmer's dashboard once you have your PR ready 💪</details>

**Comment by opiredev** [comment]
@nabby27 created a $30.00 reward using [Opire](https://opire.dev) 
 <details><summary>How to earn this reward?</summary>

Since this project does not have Opire bot installed yet 😞 you need to go to [Opire](https://app.opire.dev) and claim the rewards through your programmer's dashboard once you have your PR ready 💪</details>

**Comment by swnck** [comment]
Can you explain the Issue more?

**Comment by radumarias** [comment]
@swnck check it now. Let me know if you want to work on it so I add you to the project and to slack.

**Comment by radumarias** [comment]
@swnck is this this clearer now? let me know if more explanation is needed. Btw, how are you with this? Managed to investigate if the suggested solution could work for us?

**Comment by dmikushin** [comment]
The two most well-known FUSE-based FS are SSHFS and OverlayFS. I agree that [winfsp](https://github.com/winfsp/winfsp) - a FUSE-like backend for Windows - appears to be promising. The [sshfs-win](https://github.com/winfsp/sshfs-win) implementation is based upon it. Perhaps, some Rust fan should create bindings for winfsp, so that it could be evaluated as a technical solution for your task setting.

**Comment by radumarias** [comment]
@dmikushin thanks a lot, would you be interested to contribute to this?

**Comment by dmikushin** [comment]
@radumarias , I think the key prerequisite for this task is to provide a native Windows build of rencfs in principle. Currently, you support Windows via WSL. But in order to have the filesystem integration with Windows, the rencfs backend must compile in Windows natively.  This aspect is a hidden complexity, which does not belong to this task and should be split out as a prerequisite.

**Comment by radumarias** [comment]
@dmikushin build on windows should work, it will have a dummy implementation for FUSE that does nothing. can you give it a try and built it and let me know if there are some issues

**Comment by dmikushin** [comment]
Please note: wincfp is GPLv3, which is [not compatible](https://www.apache.org/licenses/GPL-compatibility.html) with Apache-licensed rencfs. IANAL, but to move ahead from PoC you may need to re-license rencfs as GPLv3, or collaborate with winfsp on commercial basis.

**Comment by radumarias** [comment]
https://winfsp.dev/ you talk about this?

**Comment by dmikushin** [comment]
Yes, it's the same software you've mentioned in the first post here.

**Comment by Daltonray625** [comment]
@radumarias — is any separately paid, bounded documentation or regression-verification scope still available around #3? I see the pending implementations in #322/#327 and the Opire mapping problem reported in #327; I would not duplicate those ports or assume entitlement to their reward.

I could check one specific gap you select using Linux/WSL and public test inputs, returning reproducible commands, actual results and explicit platform limitations. I am not offering native Windows-driver validation. Before any work, could you confirm whether that would be useful, the acceptance criterion, fee and payment route, and whether issue-thread scoping is acceptable before the documented project/Slack onboarding? No claim or reservation is being made.
